drivers/display/display_ili9806e_dsi.c

```diff
@@ -0,0 +1,440 @@
+	{.reg = 0x17, .cmd_len = 1, .cmd = {0x00}},
+	{.reg = 0x18, .cmd_len = 1, .cmd = {0x00}},
+	{.reg = 0x19, .cmd_len = 1, .cmd = {0x00}},
+	{.reg = 0x1a, .cmd_len = 1, .cmd = {0x00}},
+	{.reg = 0x1b, .cmd_len = 1, .cmd = {0x00}},
+	{.reg = 0x1c, .cmd_len = 1, .cmd = {0x00}},
+	{.reg = 0x1d, .cmd_len = 1, .cmd = {0x00}},
+	/* GIP Control 2 */
+	{.reg = 0x20, .cmd_len = 1, .cmd = {0x01}},
+	{.reg = 0x21, .cmd_len = 1, .cmd = {0x23}},
+	{.reg = 0x22, .cmd_len = 1, .cmd = {0x45}},
+	{.reg = 0x23, .cmd_len = 1, .cmd = {0x67}},
+	{.reg = 0x24, .cmd_len = 1, .cmd = {0x01}},
+	{.reg = 0x25, .cmd_len = 1, .cmd = {0x23}},
+	{.reg = 0x26, .cmd_len = 1, .cmd = {0x45}},
+	{.reg = 0x27, .cmd_len = 1, .cmd = {0x67}},
+	/* GIP Control 3 */
+	{.reg = 0x30, .cmd_len = 1, .cmd = {0x01}},
+	{.reg = 0x31, .cmd_len = 1, .cmd = {0x11}},
+	{.reg = 0x32, .cmd_len = 1, .cmd = {0x00}},
+	{.reg = 0x33, .cmd_len = 1, .cmd = {0xee}},
+	{.reg = 0x34, .cmd_len = 1, .cmd = {0xff}},
+	{.reg = 0x35, .cmd_len = 1, .cmd = {0xcb}},
+	{.reg = 0x36, .cmd_len = 1, .cmd = {0xda}},
+	{.reg = 0x37, .cmd_len = 1, .cmd = {0xad}},
+	{.reg = 0x38, .cmd_len = 1, .cmd = {0xbc}},
+	{.reg = 0x39, .cmd_len = 1, .cmd = {0x76}},
+	{.reg = 0x3a, .cmd_len = 1, .cmd = {0x67}},
+	{.reg = 0x3b, .cmd_len = 1, .cmd = {0x22}},
+	{.reg = 0x3c, .cmd_len = 1, .cmd = {0x22}},
+	{.reg = 0x3d, .cmd_len = 1, .cmd = {0x22}},
+	{.reg = 0x3e, .cmd_len = 1, .cmd = {0x22}},
+	{.reg = 0x3f, .cmd_len = 1, .cmd = {0x22}},
+	{.reg = 0x40, .cmd_len = 1, .cmd = {0x22}},
+	/* GOUT VGLO Control */
+	{.reg = 0x53, .cmd_len = 1, .cmd = {0x10}},
+	{.reg = 0x54, .cmd_len = 1, .cmd = {0x10}},
+	/* Change to Page 7 CMD for Normal command */
+	{.reg = 0xff, .cmd_len = 5, .cmd = {0xff, 0x98, 0x06, 0x04, 0x07}},
+	/* VREG1/2OUT ENABLE */
+	{.reg = 0x18, .cmd_len = 1, .cmd = {0x1d}},
+	{.reg = 0x26, .cmd_len = 1, .cmd = {0xb2}},
+	{.reg = 0x02, .cmd_len = 1, .cmd = {0x77}},
+	{.reg = 0xe1, .cmd_len = 1, .cmd = {0x79}},
+	{.reg = 0x17, .cmd_len = 1, .cmd = {0x22}},
+	/* Change to Page 0 CMD for Normal command */
+	{.reg = 0xff, .cmd_len = 5, .cmd = {0xff, 0x98, 0x06, 0x04, 0x00}}};
+
+static int ili9806e_write_reg(const struct device *dev, uint8_t reg, const uint8_t *buf, size_t len)
+{
+	int ret;
+	const struct ili9806e_config *cfg = dev->config;
+
+	ret = mipi_dsi_dcs_write(cfg->mipi_dsi, cfg->channel, reg, buf, len);
+	if (ret < 0) {
+		LOG_ERR("Failed writing reg: 0x%x result: (%d)", reg, ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int ili9806e_write_reg_val(const struct device *dev, uint8_t reg, uint8_t value)
+{
+	return ili9806e_write_reg(dev, reg, &value, 1);
+}
+
+static int ili9806e_write_sequence(const struct device *dev, const struct ili9806e_init_cmd *cmd,
+				   uint8_t nr_cmds)
+{
+	int ret = 0;
+
+	/* Loop trough all commands as long as writes are successful*/
+	for (int i = 0; i < nr_cmds && ret == 0; i++) {
+		ret = ili9806e_write_reg(dev, cmd->reg, cmd->cmd, cmd->cmd_len);
```

Comment:
Check `ret` here, and return if it is nonzero.

Reply:
I fixed it